source/grid/grid_generator.cc

```diff
@@ -3444,13 +3444,13 @@ namespace GridGenerator
         // grid around the cylinder to the new TFI manifold id.
         if (cell->manifold_id() == tfi_manifold_id)
           {
-            for (const unsigned int face_n : GeometryInfo<2>::face_indices())
+            for (const unsigned int face_n : cell->face_indices())
               {
                 const auto &face = cell->face(face_n);
                 if (face->at_boundary() &&
                     internal::point_in_2d_box(face->center(),
                                               center,
-                                              outer_radius))
+                                              outer_radius * (1. - 1e-12)))
```

Comment:
For context: We have a domain that is a square on the outside and a circle on the inside, with 8 cells sitting in the gap. We want to attach a polar manifold to the circle on the inside, so we ask whether we are looking at a face that is (i) at the boundary, and (ii) on the *inside* of the box. This should only match the faces on the circle, but due to changes in round-off, one of the eight faces on the outside also matched the test, and we attached the wrong manifold to it. By using a slightly smaller box, we avoid this.